### ISSUE TITLE
System.Index throwing expression tree may not contain range expression since 16.6.1

### ISSUE THREAD

**Issue by AcidRaZor** [dotnet/roslyn#44970]
**Version Used**: VS.NET 16.6.1

I have working code in .NET Core 3.1 that is now refusing to build after VS.NET update this morning.

Error	CS8792	An expression tree may not contain a range ('..') expression.		
Error	CS8790	An expression tree may not contain a pattern System.Index or System.Range indexer access	

**Steps to Reproduce**:

Index start_WriteLine1 = 2; Index end_WriteLine1 = 12;
            Index start_WriteLine2 = 12; Index end_WriteLine2 = 22;
            Index start_WriteLine3 = 22; Index end_WriteLine3 = 32;
            Index start_WriteLine4 = 32; Index end_WriteLine4 = 42;
            Index start_WriteLine5 = 42; Index end_WriteLine5 = 52;
            Index start_WriteLine6 = 52; Index end_WriteLine6 = 62;

            CreateMap<ushort[], RS485WriteBlock>().ConstructUsing(
                 data => new RS485WriteBlock
                 {
                     DataLength = data[0],
                     DestinationAddress = data[1],
                     WriteLine1 = data[start_WriteLine1..end_WriteLine1].Aggregate("", (current, next) => current + "" + next.ToString("X").ConvertHexToASCII()),
                     WriteLine2 = data[start_WriteLine2..end_WriteLine2].Aggregate("", (current, next) => current + "" + next.ToString("X").ConvertHexToASCII()),
                     WriteLine3 = data[start_WriteLine3..end_WriteLine3].Aggregate("", (current, next) => current + "" + next.ToString("X").ConvertHexToASCII()),
                     WriteLine4 = data[start_WriteLine4..end_WriteLine4].Aggregate("", (current, next) => current + "" + next.ToString("X").ConvertHexToASCII()),
                     WriteLine5 = data[start_WriteLine5..end_WriteLine5].Aggregate("", (current, next) => current + "" + next.ToString("X").ConvertHexToASCII()),
                     WriteLine6 = data[start_WriteLine6..end_WriteLine6].Aggregate("", (current, next) => current + "" + next.ToString("X").ConvertHexToASCII()),
                     WriteSequenceNumber = data[63],
                     AcknowledgeSequenceNumber = data[64]
                 }
             );

**Expected Behavior**:

To build as it did 24 hours ago and work

**Actual Behavior**:

Not building

I tried modifying/reinstalling .net Core 3.1 / cleaning the solution etc, but can't get it to build. I included working code that was index splicing an array of ushort values perfectly fine until today with no modifications to code other than to try and rebuild to deploy to production

**Comment by agocke** [comment]
Yup, it looks like there are certain cases where the Index and Range are not present in the return type of the delegate where this compiled correctly.

**Comment by agocke** [comment]
There's nothing wrong with these operators working in principle, but we have to make sure to implement and test it in all circumstances before removing the error.
